api/PayrollProcessor.Web.Api/Features/Employees/EmployeePayrollCreate.cs

```diff
@@ -43,11 +44,13 @@ public class EmployeePayrollCreate : BaseAsyncEndpoint<EmployeePayrollCreateRequ
                 .Dispatch(new EmployeeQuery(request.EmployeeId))
                 .Bind(employee =>
                 {
+                    string payrollPeriod = (ISOWeek.GetWeekOfYear(request.CheckDate.DateTime) / 2).ToString().PadLeft(2, '0');
```

Comment:
@seangwright I felt it was more intuitive to let the api determine the payroll period using a check date passed from the client. We currently use the week of the year but in theory there could be an offset if the starting week occurred mid year. This is taken from [this](https://github.com/KyleMcMaster/payroll-processor/blob/master/api/PayrollProcessor.Infrastructure.Seeding/Features/Generators/EmployeeSeed.cs) seed code, is there a shared library we could move this to, or do you think this is fine to duplicate for now?

Reply:
I say leave it and open an issue to look at converting `EmployeePayrollNew` to a closed domain model with required constructor params.

Then `newPayroll.Period` could be a readonly prop that does the above calculation/transformation.